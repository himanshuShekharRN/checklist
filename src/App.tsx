import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
