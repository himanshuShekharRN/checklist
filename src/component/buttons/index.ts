import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
