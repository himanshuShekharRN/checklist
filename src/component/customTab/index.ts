import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
