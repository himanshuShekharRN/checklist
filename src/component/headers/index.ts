import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
