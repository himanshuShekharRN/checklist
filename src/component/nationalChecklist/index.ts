import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
