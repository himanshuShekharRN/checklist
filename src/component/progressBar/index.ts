import {LinearProgressBar} from './linearProgressBar';

export {LinearProgressBar};
