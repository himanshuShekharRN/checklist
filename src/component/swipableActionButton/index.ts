import {SwipableActionButton} from './SwipableActionButton';

export {SwipableActionButton};
