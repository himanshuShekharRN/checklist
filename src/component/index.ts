import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
