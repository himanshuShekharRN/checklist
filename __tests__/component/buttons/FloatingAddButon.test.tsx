import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';

import {FloatingAddButton} from '../../../src/component';

describe('Floating Add Button', () => {
  const onPressAddHandlerMock = jest.fn();
  const customContainerStyle = {
    backgroundColor: 'transparent',
    borderRadius: 20,
  };

  const floatingButtonProps = {
    testID: 'floatingBtn',
    onPressAddHandler: onPressAddHandlerMock,
    containerStyle: customContainerStyle,
  };

  const tree: any = render(<FloatingAddButton {...floatingButtonProps} />);

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('render with a Plus icon', () => {
    render(<FloatingAddButton {...floatingButtonProps} />);

    const button = screen.getByTestId('plusIcon');
    expect(button).toBeTruthy();
  });

  test('calls onPress when the button is pressed', () => {
    render(<FloatingAddButton {...floatingButtonProps} />);

    const button = screen.getByTestId('floatingBtn');
    fireEvent.press(button);

    expect(onPressAddHandlerMock).toHaveBeenCalled();
  });

  test('applies custom containerStyle', () => {
    render(<FloatingAddButton {...floatingButtonProps} />);

    const button = screen.getByTestId('floatingBtn');
    expect(button.props.style).toContainEqual(customContainerStyle);
  });

  test('snapshot created successfully', () => {
    render(<FloatingAddButton {...floatingButtonProps} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
