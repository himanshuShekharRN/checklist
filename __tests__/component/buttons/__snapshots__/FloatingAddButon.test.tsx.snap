// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floating Add Button snapshot created successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "bottom": 50,
        "position": "absolute",
        "right": 16,
      },
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
      },
    ]
  }
  testID="floatingBtn"
>
  <SvgMock
    testID="plusIcon"
  />
</View>
`;
